### ISSUE TITLE
Add better spacing to the headers in our applications

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#156]

**Comment by benteSSB** [comment]
Døme frå SSB som viser korleis me ikkje ønskjer å ha det (same avstand mellom element som høyrer saman, som mellom element som bør vere meir avskilde): 
![image](https://github.com/Altinn/app-frontend-react/assets/77789224/a7f11f30-ee25-4713-8929-c8c442d10f1b)

**Comment by Febakke** [comment]
@benteSSB 
Ja, det så rart ut. Tenker at label - beskrivelse - skjemakomponent skal henge mer sammen ja. Skal sjekke opp dette.

**Comment by adamhaeger** [comment]
@olavflar, @Edavda and me had a discussion about this today. We agreed that:

- Designers will spec the desired margins internally in components
- We will consider implementing the ability turn on margin for certain components (ie headers) to cover edge cases.

@olavflar, @Edavda will discuss this with the designsystem team.

**Comment by Edavda** [comment]
Summary after discussion with @Febakke and @olavflar:

- Text field components should have an option to toggle a predefined top margin above the component. We might have to test out some different spacing options, but a good suggestion is `spacing #5 (22,5 px)` for medium components.
- Add options to toggle top and bottom margin for headers. Spacing of margin depends on the size of the header. 

[Link to figma](https://www.figma.com/file/Pvti3aRcOwH0k5Z7DrLznk/Arbeidsomr%C3%A5de%3A-Altinn-Studio-APPS?type=design&node-id=7893-121866&mode=design&t=hXraieBPxVv4QgfX-4)

![Image](https://github.com/Altinn/app-frontend-react/assets/2195512/2e6cdc78-2baf-4242-985e-7a5c918c9a28)

**Comment by anderssonsaja** [comment]
Hei! Jobbes det med en mulighet for våre egne utviklere til å styre dette, eller skal det settes fast på komponenter eller egen "spacing-komponent"? Hva er status på dette nå? :)

**Comment by xmrsa** [comment]
This is a typical example of the problem: 

![Image](https://github.com/user-attachments/assets/672a32ca-5059-4b26-8143-7ceaedeeaf95)

To make the layout better, I tried putting a horizontal rule `<hr>` in a paragraph-element after the first repeating group. The result is more space, but it looks like you have a dirty spot on your screen instead of a thin line. For some reason the size of the "line" is only two pixels. 

![Image](https://github.com/user-attachments/assets/1d75aa62-6160-4a64-915c-d8f944cd490d)

**Comment by xmrsa** [comment]
I see that the last frontend version has a new "type": "Divider", that makes a horizontal rule. That is good news.🥳

**Comment by anderssonsaja** [comment]
If using <hr> element, just make sure to add the aria attribute aria-hidden="true" or equivalent to make sure that it is not readable using a screen reader :)
